chain/client/src/stateless_validation/state_witness_producer.rs

```diff
@@ -49,6 +50,7 @@ pub(crate) struct CreateWitnessResult {
     pub(crate) state_witness: ChunkStateWitness,
     /// Contracts accessed and deployed while applying the chunk.
     pub(crate) contract_updates: ContractUpdates,
+    pub(crate) main_transition_shard_id: ShardId,
```

Comment:
maybe add some comment here, especially because one could wonder why we do not fetch the shard-id from the state witness instead of explicitly storing the shard_id

Reply:
good point!